rusk/src/lib/chain/rusk.rs

```diff
@@ -147,9 +149,15 @@ impl Rusk {
                     block_gas_left -= gas_spent;
                     let gas_price = unspent_tx.inner.fee.gas_price;
                     dusk_spent += gas_spent * gas_price;
+                    let economic_gas_spent = match receipt.economic_mode {
+                        Allowance(allowance) if allowance > 0 => 0,
+                        Charge(charge) if charge > 0 => charge,
+                        _ => gas_spent,
+                    };
                     spent_txs.push(SpentTransaction {
                         inner: unspent_tx,
                         gas_spent,
+                        economic_gas_spent,
```

Comment:
A spent transaction should have the whole economic mode, otherwise we cannot know if a tx was a Allowance or Charge

Reply:
if zero, it means allowance was applied, if greater than gas_spent then this is charge, if equal to gas_spent, means no economic mode - but let's consider if we could just return EconomicMode, I am open to that, although this also makes sense and it is easer to reason with IMO